src/shogun/machine/BaggingMachine.cpp

```diff
@@ -44,17 +47,53 @@ CBaggingMachine::~CBaggingMachine()
 
 CBinaryLabels* CBaggingMachine::apply_binary(CFeatures* data)
 {
-	SGVector<float64_t> combined_vector = apply_get_outputs(data);
+	SGMatrix<float64_t> output = apply_outputs_without_combination(data);
+
+	CMeanRule* mean_rule = new CMeanRule();
+
+	SGVector<float64_t> labels = m_combination_rule->combine(output);
+	SGVector<float64_t> probabilities = mean_rule->combine(output);
+
+	float64_t threshold = -0.5;
+	CBinaryLabels* pred = new CBinaryLabels(labels, threshold);
 
-	CBinaryLabels* pred = new CBinaryLabels(combined_vector);
 	return pred;
 }
 
 CMulticlassLabels* CBaggingMachine::apply_multiclass(CFeatures* data)
 {
-	SGVector<float64_t> combined_vector = apply_get_outputs(data);
+	SGMatrix<float64_t> bagged_outputs = apply_outputs_without_combination(data);
+
+	int32_t num_samples = bagged_outputs.size() / m_num_bags;
+	int32_t num_classes = dynamic_cast<CMulticlassLabels*>(m_labels)->get_num_classes();
```

Comment:
this cast is not safe, so you need to store the casted pointer in a variable and then do a 
`REQUIRE(labels_multiclass, "Labels are not multiclass\n");`
or even better, try to convert them to multiclass (there are constructors to e.g. turn a binary labels into multiclass), and then say "Could not convert labels to multiclass"